src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentLoaderService.cs

```diff
@@ -24,21 +25,26 @@ public PdbSourceDocumentLoaderService()
         {
         }
 
-        public Task<TextLoader?> LoadSourceDocumentAsync(SourceDocument sourceDocument, CancellationToken cancellationToken)
+        public Task<TextLoader?> LoadSourceDocumentAsync(SourceDocument sourceDocument, Encoding? defaultEncoding, CancellationToken cancellationToken)
         {
             // If we already have the embedded text then use that directly
-            if (sourceDocument.EmbeddedText is not null)
+            SourceText? sourceText = null;
+            if (sourceDocument.EmbeddedTextBytes is not null)
             {
-                var textAndVersion = TextAndVersion.Create(sourceDocument.EmbeddedText, VersionStamp.Default, sourceDocument.FilePath);
-                return Task.FromResult<TextLoader?>(TextLoader.From(textAndVersion));
+                sourceText = TryLoadSourceFromEmbeddedSource(sourceDocument.EmbeddedTextBytes, defaultEncoding);
             }
 
             // Otherwise, check the easiest (but most unlikely) case which is the document exists on the disk
-            if (File.Exists(sourceDocument.FilePath))
+            if (sourceText is null && File.Exists(sourceDocument.FilePath))
```

Comment:
When we fail `TryLoadSourceFromEmbeddedSource` should we still look at loading from a file? Are there cases where that is correct? 

I'm guessing we could basically have 

```c#
var sourceText = sourceDocument.EmbeddedTextBytes is not null
    ? TryLoadSourceFromEmbeddedSource(sourceDocument.EmbeddedTextBytes, defaultEncoding)
    : TryLoadFromFile(sourceDocument) // This would be an added method 
```

Reply:
Realistically it probably doesn't make sense, but in my mind the lookup is just a series of locations we check in turn, embedded, on disk, from the debugger (network). There probably isn't a high chance that the embedded source would be corrupt, but the on disk source would still match the checksum etc., but from a purely code style point of view I don't like ternaries.

Reply:
And in a complete backflip, I've refactored to almost this 😁